File: Python/DecryptAffine_German.py
from ExtendedEuclidean import ExtendedEuclidean


def DecryptAffine_German(ciphertext_ascii_30, a, b):
    t = ExtendedEuclidean(30, a)["t"]
    return [(((t * (i - 65) - b * t) % 30) + 97) for i in ciphertext_ascii_30]

File: R/DecryptAffine_German.r
source('ExtendedEuclidean.r')

DecryptAffine_German <- function(ciphertext_ascii_dec, a, b)
{
  ext_eucl <- ExtendedEuclidean(30, a)
  return(((
    ext_eucl[['t']] * (ciphertext_ascii_dec - 65) - b
  ) %% 30) + 97)
}